utils/resultwriter.go

```diff
@@ -122,24 +129,33 @@ func (rw *ResultsWriter) printScanResultsTables() (err error) {
 		printMessage(coreutils.PrintTitle("The full scan results are available here: ") + coreutils.PrintLink(resultsPath))
 	}
 	log.Output()
-	if rw.includeVulnerabilities {
-		err = PrintVulnerabilitiesTable(vulnerabilities, rw.results, rw.isMultipleRoots, rw.printExtended, rw.scanType)
-	} else {
-		err = PrintViolationsTable(violations, rw.results, rw.isMultipleRoots, rw.printExtended, rw.scanType)
-	}
-	if err != nil {
-		return
+	if slices.Contains(rw.subScansPreformed, ScaScan) {
+		if rw.includeVulnerabilities {
+			err = PrintVulnerabilitiesTable(vulnerabilities, rw.results, rw.isMultipleRoots, rw.printExtended, rw.scanType)
+		} else {
+			err = PrintViolationsTable(violations, rw.results, rw.isMultipleRoots, rw.printExtended, rw.scanType)
+		}
+		if err != nil {
+			return
+		}
+		if rw.includeLicenses {
+			if err = PrintLicensesTable(licenses, rw.printExtended, rw.scanType); err != nil {
+				return
+			}
+		}
 	}
-	if rw.includeLicenses {
-		if err = PrintLicensesTable(licenses, rw.printExtended, rw.scanType); err != nil {
+	if slices.Contains(rw.subScansPreformed, SecretsScan) {
+		if err = PrintSecretsTable(rw.results.ExtendedScanResults.SecretsScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
 			return
 		}
 	}
-	if err = PrintSecretsTable(rw.results.ExtendedScanResults.SecretsScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
-		return
+	if slices.Contains(rw.subScansPreformed, IacScan) {
+		if err = PrintIacTable(rw.results.ExtendedScanResults.IacScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
+			return
+		}
 	}
-	if err = PrintIacTable(rw.results.ExtendedScanResults.IacScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
-		return
+	if !slices.Contains(rw.subScansPreformed, SastScan) {
+		return nil
 	}
 	return PrintSastTable(rw.results.ExtendedScanResults.SastScanResults, rw.results.ExtendedScanResults.EntitledForJas)
```

Comment:
1. any reason to use ! Contains with nil, as oppose to the other that check "Contains" ?
2. BTW - why does the PrintSastTable is different that others? (doesn't return err)

Reply:
PrintSastTable does return err, its the end of the func...